bindings/CMakeLists.txt

```diff
@@ -5,6 +5,10 @@
 # For the list of contributors see $ROOTSYS/README/CREDITS.
 
 if(pyroot)
+  if (NOT CMAKE_INSTALL_LIBDIR STREQUAL CMAKE_INSTALL_PYTHONDIR AND ((how_many_pythons GREATER 1) OR MSVC OR pyroot_legacy))
+    message(FATAL_ERROR "Using separate CMAKE_INSTALL_PYTHONDIR and CMAKE_INSTALL_LIBDIR is not supported for multi-python builds, on Windows or if pyroot_legacy=ON. Please check your build configuration.")
```

Comment:
Please make it work with legacy pyroot as well at least, as it's much simpler than for new PyROOT (requires only a single-line change in RootMacros.cmake).

Reply:
I think the changes to RootMacros.cmake that are required to make legacy PyROOT work are already contained in this PR?

However, we will not test/exercise the feature with legacy PyROOT, and this being a new feature I'm not convinced I should spend time making sure it works for legacy PyROOT too? (note that I am dedicating time to PyROOT+build system mainly because I am trying to make root.deb happen).

Reply:
If things only work for specific configurations, users will get annoyed and file bugs in any case. If just a single line change is needed (I can test wether it works or not), then it's worth the extra effort. Making the DEB package work is the goal, but if that's done by addressing symptoms rather than causes, it's work that will very likely need to be redone later.

Reply:
I think we agree that (once the other two threads are resolved) this PR will be making the build system strictly better than before -- so I don't think the fact that old PyROOT is left out is a good enough reason to not merge it.

With that said, if you can indicate what the single-line change that's missing is (and it's in our build system and not a patch to cppyy) I would be happy to add it to the PR. You are also welcome to push to this branch directly.

Reply:
I saw later that the line that's necessary for old PyROOT is included, so if you just don't send a `FATAL_ERROR` when `pyroot_legacy` is enabled, it should be enough for it to work. Right now, with old PyROOT your `if` from above will not let users use this feature with old PyROOT.

Reply:
ok perfect, i'll change the condition so old PyROOT does not trigger a fatal error :+1:

Reply:
old PyROOT is now allowed, can you please check that it works? :smile:

Reply:
@etejedor @amadio I tested with legacy PyROOT. `libPyROOT.so` is installed in `CMAKE_ROOT_LIBDIR` and it can't be found from python at `import ROOT` time.

For legacy PyROOT we could do [this](https://github.com/root-project/root/pull/5907/files#diff-313bc1b7a87ec940b65d43218409d171) and just tell `ROOT.py` to also look in `root-config --libdir`, or we can add another symlink.

Let me know what you prefer.

Reply:
To be consistent I would add a symlink, the alternative you suggested can also work but it would need to be applied to cppyy too (otherwise only `import ROOT` would work, but not `import cppyy`).

Reply:
should be fixed